nilearn/image/resampling.py

```diff
@@ -620,8 +509,6 @@ def resample_img(
         dtype=resampled_data_dtype,
     )
 
-    all_img = (slice(None),) * 3
-
     # if (A == I OR some combination of permutation(I) and sign-flipped(I)) AND
     # all(b == integers):
     if (
```

Comment:
here also we might want 2 helper functions for the case where it is just a question of indexing (if) and the case where actual resampling takes place (else)?